### ISSUE TITLE
Recognizing session type failed with Qt 5.12.8

### ISSUE THREAD

**Issue by fancyfarmer** [Skycoder42/QHotkey#74]
Trying to link library with another project gives:
```
undefined reference to `QX11Info::isPlatformX11()'
```
I've noticed that in doc it shall be included in Qt 5.2:
https://doc.qt.io/qt-5/qx11info.html#isPlatformX11

but it looks like it is not. It was included in following commit:
https://github.com/qt/qtbase/commit/edceff30b4093e2bef6e80c7552edf595134f351
which precedes: v6.2.0-alpha1, v6.3.0-alpha1, v6.4.0-beta1

therefore check shall looks like:
```
#if QT_VERSION >= QT_VERSION_CHECK(6, 2, 0)
	return qGuiApp->nativeInterface<QNativeInterface::QX11Application>();
#else
	char *v = getenv("XDG_SESSION_TYPE");
	return (v == NULL || strcasecmp(v, "x11") == 0);
#endif
```

Similar issue can be recognized with:
```
undefined reference to `QX11Info::display()'
```
but I haven't take a look closely on it yet.

**Comment by fancyfarmer** [comment]
Maybe I'm making a mistake here somewhere but I've checked sources od 5.12.8 and 5.2 and there is not much related to "qtx11" string, 
```
qtbase-5.12.8$ grep -rl qtx11 .
./qmake/doc/qmake.qdocconf
```
and
```
qtbase-5.2.0$ grep -rl qtx11 .
./qmake/doc/qmake.qdocconf
```
No findings in .cpp or ,h files at all.

**Comment by Shatur** [comment]
Perhaps you built without QtExtras? Because I [have](https://archlinux.org/packages/extra/x86_64/qt5-x11extras/) these files:
```
    usr/
    usr/include/
    usr/include/qt/
    usr/include/qt/QtX11Extras/
    usr/include/qt/QtX11Extras/QX11Info
    usr/include/qt/QtX11Extras/QtX11Extras
    usr/include/qt/QtX11Extras/QtX11ExtrasDepends
    usr/include/qt/QtX11Extras/QtX11ExtrasVersion
    usr/include/qt/QtX11Extras/qtx11extrasglobal.h
    usr/include/qt/QtX11Extras/qtx11extrasversion.h
    usr/include/qt/QtX11Extras/qx11info_x11.h
    usr/lib/
    usr/lib/cmake/
    usr/lib/cmake/Qt5X11Extras/
    usr/lib/cmake/Qt5X11Extras/Qt5X11ExtrasConfig.cmake
    usr/lib/cmake/Qt5X11Extras/Qt5X11ExtrasConfigVersion.cmake
    usr/lib/libQt5X11Extras.prl
    usr/lib/libQt5X11Extras.so
    usr/lib/libQt5X11Extras.so.5
    usr/lib/libQt5X11Extras.so.5.15
    usr/lib/libQt5X11Extras.so.5.15.5
    usr/lib/pkgconfig/
    usr/lib/pkgconfig/Qt5X11Extras.pc
    usr/lib/qt/
    usr/lib/qt/mkspecs/
    usr/lib/qt/mkspecs/modules/
    usr/lib/qt/mkspecs/modules/qt_lib_x11extras.pri
    usr/lib/qt/mkspecs/modules/qt_lib_x11extras_private.pri
    usr/share/
    usr/share/licenses/
    usr/share/licenses/qt5-x11extras
```

**Comment by fancyfarmer** [comment]
$ dpkg -L libqt5x11extras5
/.
/usr
/usr/lib
/usr/lib/x86_64-linux-gnu
/usr/lib/x86_64-linux-gnu/libQt5X11Extras.so.5.12.8
/usr/share
/usr/share/doc
/usr/share/doc/libqt5x11extras5
/usr/share/doc/libqt5x11extras5/changelog.Debian.gz
/usr/share/doc/libqt5x11extras5/copyright
/usr/lib/x86_64-linux-gnu/libQt5X11Extras.so.5
/usr/lib/x86_64-linux-gnu/libQt5X11Extras.so.5.12

**Comment by Shatur** [comment]
So looks like you have it too?

**Comment by fancyfarmer** [comment]
That's right together with dev headers. It should just work, but mu point is that there is no method called isPlatformX11 in QtX11extrass lib itself until 6.2. Therefore it looks like we cannot use it in QHotKey code whenever Qt version <=6.2.

**Comment by Shatur** [comment]
> That's right together with dev headers. It should just work, but mu point is that there is no method called isPlatformX11 in QtX11extrass lib itself until 6.2

That's strange. Because I use it with Qt5 on my machine for several years for sure. I believe that something wrong with linkage on your machine.

**Comment by Shatur** [comment]
So QHotkey should link with X11Extras for Qt5?

**Comment by fancyfarmer** [comment]
Ok, you are right. I've even found it in CMakeList
```
if(QT_DEFAULT_MAJOR_VERSION EQUAL 6)
    set(CPACK_DEBIAN_PACKAGE_DEPENDS "libqt6x11extras6 (>= 6.2.0)")
else()
    set(CPACK_DEBIAN_PACKAGE_DEPENDS "libqt5x11extras5 (>= 5.15.2)")
endif()
```

but I still have to dig some deeper into this why I'm getting 
```
undefined reference to `QX11Info::isPlatformX11()
```
then. I need some time to investigate. Thx

**Comment by Shatur** [comment]
I believe we link it here:

https://github.com/Skycoder42/QHotkey/blob/34777f2e91b72cdc62a415975cc3ec53475ed022/CMakeLists.txt#L62-L66

But it doesn't work for you?

**Comment by fancyfarmer** [comment]
In the way I wan't to it doesn't work. What I've done is including following line to my cmake:
```
find_package(QHotkey                        REQUIRED)
```
but I've got following:
```
... links to target "Qt5::X11Extras" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```
That's indicating that library dependency is propagating from qhotkey library to my project, that's well. 
In my cmake I've used
```
get_target_property(OUT qhotkey INTERFACE_LINK_LIBRARIES)
message(STATUS ${OUT} "-----INTERFACE_LINK_LIBRARIES-----")
```
to determine propagation fact and correctly got:
```
Qt5::CoreQt5::GuiQt5::X11Extras
```

The point is unless I add:
```
find_package(Qt5X11Extras           REQUIRED)
```
to my cmake I am getting the error like at the beginning. After adding this find_package line (and another one - not important right now) - cmake generates correctly and project is built and linked. The question is if that is proper way of 'cmake`ing' when main project has finally to include dependencies related to libraries that are linked together with it.

Any idea what I'm doing wrong here?
Thanks

**Comment by Shatur** [comment]
> links to target "Qt5::X11Extras" but the target was not

So CMake cannot find `Qt5::X11Extras` on your system. Unfortunately, something wrong with your system configuration.

**Comment by fancyfarmer** [comment]
This error is when there is no find_package in my cmake. When find_package is used then all is fine. The question is: shall I specify this explicitly in my cmake. 

Note the even with other libraries used by main project I have to use e.g.:
find_package(Qt5Gui           REQUIRED)

**Comment by Shatur** [comment]
> The point is unless I add: find_package(Qt5X11Extras REQUIRED) to my cmake I am getting the error like at the beginning. 

Can't help you here. It should just work.
